src/PrivateDns/PrivateDns/Zones/NewAzurePrivateDnsZone.cs

```diff
@@ -41,6 +42,10 @@ public class NewAzurePrivateDnsZone : PrivateDnsBaseCmdlet
         public override void ExecuteCmdlet()
         {
             this.Name = TrimTrailingDotInZoneName(this.Name);
+            if (this.Name.EndsWith(".local", StringComparison.OrdinalIgnoreCase))
```

Comment:
You don't need the ```this.``` for any of these, please remove.

Can we add regression test coverage for this (a unit test would be fine)

Reply:
Added testing. I'm not removing this to keep it consistent across the whole module.